src/librustc_mir/borrow_check/nll/universal_regions.rs

```diff
@@ -490,6 +490,29 @@ impl<'cx, 'gcx, 'tcx> UniversalRegionsBuilder<'cx, 'gcx, 'tcx> {
             &bound_inputs_and_output,
             &mut indices,
         );
+
+        // Add late-bound regions which are not used in in inputs and outhout
```

Comment:
I think it'd be nice to put this into a fn, along with a somewhat more extensive comment that (at minimum) cites the issue number.

Reply:
I think a comment like this would be good:

---

Finds late-bound regions that do not appear in the parameter listing and adds them to the indices vector. Typically, we identify late-bound regions as we process the inputs and outputs of the closure/function. However, sometimes there are late-bound regions which do not appear in the fn parameters but which are nonetheless in scope. The simplest case of this are unused functions, like `fn foo<'a>() { }` (see eg., #51351). Despite not being used, users can still reference these regions (e.g., `let x: &'a u32 = &22;`), so we need to create entries for them and store them in the `indices` map. This code iterates over the complete set of late-bound regions and checks for any that we have not yet seen, adding them to the inputs vector.